tests/appsec/iast/_test_iast_fixtures.py

```diff
@@ -253,6 +258,28 @@ def test_source_reported(self):
         for request in self.requests.values():
             self.validate_request_reported(request)
 
+    def check_test_telemetry_should_execute(self):
+        interfaces.library.assert_iast_implemented()
+
+        # to avoid false positive, we need to check that at least
+        # one test is working before running the telemetry tests
```

Comment:
Actually, I don't understand why we are doing this ¿?

Reply:
Most of the `XPASS` that we have in nodejs in IAST are because we have `missing_feature` for whole file, and for some reason the telemetry test is working, even when the endpoint is not implemented or the feature is failing. 
In those cases, failure is expected also in telemetry test, so in this method we are checking that at least one `test_` is working before checking the telemetry system test.

Reply:
Could be good, as we are now testing `test_insecure` each time that we are testing `test_secure`. But it is a huge change (I can see `test_insecure` overrrided in at least 12 files... and it is not related with this PR as we are modifying execution of metrics tests, not the `test_insecure`.

Could it be done in a different PR?

Reply:
sure !

Reply:
In that case create an issue on the repo so we keep note of it!